bin/build_cmd.ml

```diff
@@ -215,3 +215,63 @@ let fmt =
     run_build_command ~common ~config ~request
   in
   Cmd.v (Cmd.info "fmt" ~doc ~man ~envs:Common.envs) term
+
+let doc =
+  let doc = "Build the documentation of an OCaml project" in
+  let man =
+    [ `S "DESCRIPTION"
+    ; `P
+        {|$(b,dune ocaml doc) build and open the documention of an ocaml project.|}
+    ; `Blocks Common.help_secs
+    ]
+  in
+  let info = Cmd.info "doc" ~doc ~man in
+  let term =
+    let+ common = Common.term in
+    let config = Common.init common in
+    let request (setup : Import.Main.build_system) =
+      let dir = Path.(relative root) (Common.prefix_target common ".") in
+      let open Action_builder.O in
+      let+ () =
+        Alias.in_dir
+          ~name:(Dune_engine.Alias.Name.of_string "doc")
+          ~recursive:true ~contexts:setup.contexts dir
+        |> Alias.request
+      in
+      let is_default ctx =
+        ctx |> Context.name |> Dune_engine.Context_name.is_default
+      in
+      let doc_ctx = List.find_exn setup.contexts ~f:is_default in
+      let toplevel_index_path = Dune_rules.Odoc.Paths.toplevel_index doc_ctx in
+      let absolute_toplevel_index_path =
+        Path.(toplevel_index_path |> build |> to_absolute_filename)
+      in
+      let url = Printf.sprintf {|file://%s|} absolute_toplevel_index_path in
+      let cmd =
+        let open Option.O in
+        let path = Env_path.path Env.initial in
+        let* cmd_name, args =
+          match Ocaml_config.system doc_ctx.ocaml_config with
+          | "macosx" -> Some ("open", [ "-u" ])
+          | "linux" -> Some ("xdg-open", [])
+          | "windows" -> Some ("start", [])
+          | _ -> None
+        in
+        let+ p = Bin.which ~path cmd_name in
+        ( p
+        , (* First element of argv is the name of the command. *)
+          (cmd_name :: args) @ [ url ] )
+      in
+      match cmd with
+      | Some (cmd, args) ->
+        Proc.restore_cwd_and_execve
+          (Path.to_absolute_filename cmd)
+          args ~env:Env.initial
+      | None ->
+        Printf.printf "Docs built. Index can be found here: %s\n"
```

Comment:
Rather than `Printf.printf` use `Dune_console.print`. It might also be worth printing this message in both cases, whether or not the command was found. In fact, it might be clearer to raise a user error / warning here and say that a web browser command or whatever could not be found / used so that it doesn't mysteriously fail.